src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Provider.PR.Domain.Interfaces;
+using SFA.DAS.Provider.PR.Web.Authorization;
+using SFA.DAS.Provider.PR.Web.Infrastructure;
+using SFA.DAS.Provider.PR.Web.Infrastructure.Services;
+using SFA.DAS.Provider.PR.Web.Models.AddEmployer;
+using SFA.DAS.Provider.PR.Web.Models.Session;
+
+namespace SFA.DAS.Provider.PR.Web.Controllers.AddEmployer;
+
+[Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
+
+[Route("/{ukprn}/addEmployer/searchByPaye", Name = RouteNames.AddEmployerSearchByPaye)]
+public class SearchByPayeController(IOuterApiClient _outerApiClient, ISessionService _sessionService, IValidator<SearchByPayeSubmitViewModel> _validator) : Controller
+{
+    public const string ViewPath = "~/Views/AddEmployer/SearchByPaye.cshtml";
+
+
+    [HttpGet]
+    public IActionResult Index([FromRoute] int ukprn)
+    {
+        var viewModel = GetViewModel(ukprn);
```

Comment:
Move this down so we initiate view model only if session check passes.

Reply:
ok